python/ray/data/_internal/dataset_logger.py

```diff
@@ -34,17 +34,25 @@ def __init__(self, log_name: str):
         # parameter `log_to_stdout = False`, `_logger.propagate` will be set
         # to `False` in order to prevent the root logger from writing the log
         # to stdout.
-        self._logger = logging.getLogger(f"{log_name}.logfile")
-        # We need to set the log level again when explicitly
-        # initializing a new logger (otherwise can have undesirable level).
-        self._logger.setLevel(LOGGER_LEVEL.upper())
+        self.log_name = log_name
+        self._logger = None  # initialized in self._initialize_logger()
+        self._initialize_logger()
 
-        # Add log handler which writes to a separate Datasets log file
-        # at `DatasetLogger.DEFAULT_DATASET_LOG_PATH`
+    def _initialize_logger(self):
```

Comment:
how about we return `logger` as the return value? So we don't need to think about the internal state change of this class.